homeassistant/components/srp_energy/sensor.py

```diff
@@ -68,78 +89,77 @@ async def async_update_data():
 
     coordinator = DataUpdateCoordinator(
         hass,
-        _LOGGER,
+        LOGGER,
         name="sensor",
         update_method=async_update_data,
         update_interval=MIN_TIME_BETWEEN_UPDATES,
     )
 
     # Fetch initial data so we have data when entities subscribe
-    await coordinator.async_refresh()
+    await coordinator.async_config_entry_first_refresh()
 
-    async_add_entities([SrpEntity(coordinator)])
+    async_add_entities([SrpEntity(coordinator, entry_name)])
 
 
 class SrpEntity(SensorEntity):
     """Implementation of a Srp Energy Usage sensor."""
 
+    _attr_has_entity_name = True
     _attr_attribution = ATTRIBUTION
     _attr_should_poll = False
 
-    def __init__(self, coordinator):
+    def __init__(self, coordinator, entry_name):
         """Initialize the SrpEntity class."""
         self._name = SENSOR_NAME
         self.type = SENSOR_TYPE
         self.coordinator = coordinator
         self._unit_of_measurement = UnitOfEnergy.KILO_WATT_HOUR
         self._state = None
+        self._entry_name = entry_name
+        unique_id = f"{entry_name}_energy_usage".lower()
 
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{DEFAULT_NAME} {self._name}"
+        LOGGER.debug("Setting entity unique_id %s", unique_id)
+        self._attr_unique_id = unique_id
+
+        self._attr_native_value = self.coordinator.data
 
     @property
-    def unique_id(self):
-        """Return sensor unique_id."""
-        return self.type
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return f"{self._entry_name} {self._name}"
 
     @property
-    def native_value(self):
+    def native_value(self) -> StateType:
         """Return the state of the device."""
-        if self._state:
-            return f"{self._state:.2f}"
-        return None
+        LOGGER.debug(
```

Comment:
We don't allow side effects in entity state properties. Please remove or move it to an update callback.